docs/adding_packages/build_and_package.md

```diff
@@ -13,24 +13,59 @@ Both methods often use build helpers to build binaries and install them into the
 
-* Except for CMake and a working build toolchain (compiler, linker, archiver, etc.), the recipe should not assume that any other build tool is installed on the user-build machine (like Meson, autotools, or pkg-config). On Windows, the recipe should not assume that a shell is available (like MSYS2). Therefore, if the build method requires additional tools, they should be added to `build_requirements()`.
+* Except for CMake, a working build toolchain (compiler, linker, archiver, etc.), and a "native generator" (`make` on *nix platforms, `mingw32-make` for MinGW, `MSBuild`/`NMake` for Visual Studio), the recipe should not assume that any other build tool is installed on the user-build machine (like Meson, autotools, or pkg-config). On Windows, the recipe should not assume that a shell is available (like MSYS2). Therefore, if the build method requires additional tools, they should be added to `build_requirements()`.
+  Tools explicitly marked as available by users through conf like `tools.gnu:make_program`, `tools.gnu:pkg_config`, `tools.microsoft.bash:path`, `tools.microsoft.bash:subsystem` should be taken into account to conditionally inject a build requirement (these conf should have precedence over build requirement equivalent hardcoded in the recipe).
 
 * It is forbidden to run other conan client commands during build. In other words, if upstream build files call conan under the hood (through `cmake-conan` for example or any other logic), this logic must be removed.
 
 * Settings from profile should be honored (`build_type`, `compiler.libcxx`, `compiler.cppstd`, `compiler.runtime` etc).
 
-* These env vars from profile should be honored and properly propagated to underlying build system during the build: `CC`, `CXX`, `CFLAGS`, `CXXFLAGS`, `LDFLAGS`.
+* These env vars from host profile (`[env]` for conan v1 recipes, `[buildenv]` for conan v2 recipes) should be honored and properly propagated to underlying build system during the build: `CC`, `CXX`, `AR` `LD`. They should be curated on the fly if underlying build system expects a specific format (no spaces in path, forward slash instead of back slash, etc).
+
+* These compiler and linker conf from host profile should be honored and propagated to underlying build system during the build:
+  * `tools.build:cflags`
+  * `tools.build:cxxflags`
+  * `tools.build:defines`
+  * `tools.build:sharedlinklags`
+  * `tools.build:exelinkflags`
+  * `tools.apple:enable_bitcode` (only if host OS is `iOS`/`watchOS`/`tvOS`)
+
+* If host OS is Apple OS family (`macOS`/`iOS`/`watchOS`/`tvOS`), `-headerpad_max_install_names` flag should be passed to linker, unless host OS is `iOS`/`watchOS`/`tvOS` and `tools.apple:enable_bitcode` is enabled.
```

Comment:
Users may want to set `LC_RPATH` afterwards during final packaging (not conan one). If you forget this flag, it makes conan shared libs unusable for this final packaging of the product.
CMake, Autotools & Meson always add this flag under the hood.

Reply:
> If you forget this flag, it makes conan shared libs unusable for this final packaging of the product.

I don't think this is entirely true, at least not since the introduction of the `@rpath` keyword which I believe happened in 2007. 

The way the runtime linker works on macOS means that when a library needs to be loaded that has an install name that starts with`@rpath` - all `LC_RPATH` values in the load tree are taken into account, including the one from the top-level executable. So typically for packaging, if all libraries already have an install name that starts with `@rpath` (which we are now enforcing) the libraries themselves do not require any further patching, and then the responsibility of making this work at runtime is on the executable, to have a valid `LC_RPATH` entry that will ensure everything works.

It would help if there was a particular packaging scenario that we are missing that actually requires having that flag enabled.

As for CMake - this is not a publicly documented behaviour. CMake performs patching of binary files when calling the install target, to avoid having to re-link them, which surely provides some performance benefit. But they could have chosen to relink the binaries at install time.  If Meson does this, I would imagine it's for a similar purpose.

However I have not seen generic autotools add this flag at all - quite the opposite, I was under the impression that libtool performed a relink to override the rpath during install.

Reply:
I'm not 100% sure that `LC_RPATH` is transitive (it's true I never add issues when LC_RPATH was set in executable only).
Yes maybe autotools (and even Meson) don't inject this flag. CMake does for sure.

Reply:
The manpages for `dyld` do indeed mention that it is transitive in the "chain of load commands", when resolving `@rpath`:

> The run path stack is built from the LC_RPATH
>               load commands in the depencency chain that lead to the current dylib load.

So if `LC_RPATH` is set in the executable (either as an absolute path, or a path relative to @executable_path) then this should work without further modifying the dylibs. It's quite typical to find .app bundles where the executable has an rpath entry for `@executable_path/../Frameworks`, and all the dylibs/frameworks inside that folder have an install name of `@rpath/xxx`. 

If we were *not* enforcing the `@rpath` install names then I would see a case for ensuring this flag is added for all binaries - but otherwise, I think the install name convention actually negates the need for having to patch dylibs altogether. And CMake does to support its own implementation of rewriting the rpath during install, but they could have gone the libtool way and relinked, so I wouldn't necessary rely on this as if it were intended nor documented behaviour.

Reply:
@jcar87 I can remove this paragraph to not block PR. It's premature to write something regarding this topic if https://github.com/conan-io/conan/issues/12694 is not clarified.